creator/extract_configs/common.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+
+def map_mondo(s):
+    MONDO_RE = r"^MONDO:([0-9]{1,7})$"
+    if re.match(MONDO_RE, s):
+        return s
+    return None
+
+
+def map_ncit(s):
+    NCIT_RE = r"^C\d+$"
+    if re.match(NCIT_RE, s):
+        return s
+    return None
+
+
+def map_icd(s):
+    ICD_RE = r"[A-TV-Z][0-9][0-9AB]\.?[0-9A-TV-Z]{0,4}"
+    if re.match(ICD_RE, s):
+        return s
+    return None
+
+
+def map_hpo(s):
+    HPO_RE = r"^HP:[0-9]{7}$"
+    if re.match(HPO_RE, s):
+        return s
+    return None
+
+
+def map_snomed(s):
+    SNOMED_RE = r"^[0-9]*$"
+    if re.match(SNOMED_RE, s):
+        return s
+    return None
+
+
+def map_uberon(s):
+    UBERON_RE = r"^UBERON:[0-9]{7}$"
+    if re.match(UBERON_RE, s):
+        return s
+    return None
```

Comment:
As requested here is some more complex matching that also coerces values to consistent output formats. This code allows truncated input numbers, input prefixes are optional and can be separated by either : or _, and outputs are given consistent prefixes and zero padding.

```suggestion
def _get_match(prefix_pattern, code_pattern, s, result_prefix, result_zeropad=0):
    if prefix_pattern:
        prefix_pattern = f"(?:{prefix_pattern}[:_])?"

    if re.compile(code_pattern).groups == 0:
        code_pattern = f"({code_pattern})"

    capture_pattern = f"^{prefix_pattern}{code_pattern}$"
    assert re.compile(capture_pattern).groups == 1

    match = re.match(capture_pattern, s.strip(), flags=re.IGNORECASE)
    if match:
        return f"{result_prefix}{match.group(1).zfill(result_zeropad)}"
    else:
        return None


def map_mondo(s):
    return _get_match(r"MONDO", r"[0-9]{1,7}", s, "MONDO:", result_zeropad=7)


def map_ncit(s):
    return _get_match(r"NCIT", r"C([0-9]{1,7})", s, "NCIT:C", result_zeropad=7)


def map_icd10(s):
    return _get_match(
        r"ICD(?:10)?", r"[A-TV-Z][0-9][A-Z0-9](?:\.[A-Z0-9]{1,4})?", s, "ICD10:"
    )


def map_hpo(s):
    return _get_match(r"HPO?", r"[0-9]{1,7}", s, "HP:", result_zeropad=7)


def map_snomed(s):
    return _get_match(r"SNOMED", r"[0-9]+", s, "SNOMED:")


def map_uberon(s):
    return _get_match("UBERON", r"[0-9]{1,7}", s, "UBERON:", result_zeropad=7)
```


```Python
def test_matching():
    codes = {
        "MONDO": (
            map_mondo,
            {
                "0000109": "MONDO:0000109",
                "0109": "MONDO:0000109",
                "109": "MONDO:0000109",
                "A": None,
                "12345678": None,
            },
        ),
        "NCIT": (
            map_ncit,
            {
                "C0000100": "NCIT:C0000100",
                "C0100": "NCIT:C0000100",
                "C100": "NCIT:C0000100",
                "1": None,
                "A": None,
                "C12345678": None,
            },
        ),
        "HP": (
            map_hpo,
            {
                "0000100": "HP:0000100",
                "0100": "HP:0000100",
                "100": "HP:0000100",
                "A": None,
                "12345678": None,
            },
        ),
        "HPO": (
            map_hpo,
            {
                "0000100": "HP:0000100",
                "0100": "HP:0000100",
                "100": "HP:0000100",
                "A": None,
                "12345678": None,
            },
        ),
        "ICD": (
            map_icd10,
            {
                "A12": "ICD10:A12",
                "A12.1": "ICD10:A12.1",
                "A13.1114": "ICD10:A13.1114",
                "A": None,
                "12345678": None,
            },
        ),
        "ICD10": (
            map_icd10,
            {
                "A12": "ICD10:A12",
                "A12.1": "ICD10:A12.1",
                "A13.1114": "ICD10:A13.1114",
                "A": None,
                "12345678": None,
            },
        ),
    }
    for prefix, (mapfunc, values) in codes.items():
        for v, res in values.items():
            assert mapfunc(v) == res
            for sep in [":", "_"]:
                assert mapfunc(prefix + sep + v) == res
```